pkg/backends/calico/routes.go

```diff
@@ -206,6 +197,16 @@ func (rg *routeGenerator) setRouteForSvc(svc *v1.Service, ep *v1.Endpoints) {
 // advertiseThisService returns true if this service should be advertised on this node,
 // false otherwise.
 func (rg *routeGenerator) advertiseThisService(svc *v1.Service, ep *v1.Endpoints) bool {
+	// do nothing if the svc is not a relevant type
+	if (svc.Spec.Type != v1.ServiceTypeClusterIP) && (svc.Spec.Type != v1.ServiceTypeNodePort) && (svc.Spec.Type != v1.ServiceTypeLoadBalancer) {
+		return false
+	}
+
+	// also do nothing if the clusterIP is empty or None
+	if len(svc.Spec.ClusterIP) > 0 && svc.Spec.ClusterIP != "None" {
```

Comment:
I think this is backwards - should be:

```
if len(svc.Spec.ClusterIP) == 0 || svc.Spec.ClusterIP == "None"
```

If the cluster IP is > 0 and not equal to None, then we want to carry on checking if we should advertise it.